src/tools/wasm-link.cc

```diff
@@ -154,17 +154,30 @@ LinkerInputBinary::~LinkerInputBinary() {
   delete[] data;
 }
 
-static Index relocate_func_index(LinkerInputBinary* binary,
-                                 Index function_index) {
+bool LinkerInputBinary::IsFunctionImport(Index index) {
+  assert(IsValidFunctionIndex(index));
+  return index < function_imports.size();
+}
+
+bool LinkerInputBinary::IsInactiveFunctionImport(Index index){
+  return IsFunctionImport(index) && !function_imports[index].active;
+}
+
+bool LinkerInputBinary::IsValidFunctionIndex(Index index) {
+  return index < function_imports.size() + function_count;
+}
+
+Index LinkerInputBinary::RelocateFuncIndex(Index function_index) {
+  assert(IsValidFunctionIndex(function_index));
   Index offset;
-  if (function_index >= binary->function_imports.size()) {
+  if (function_index >= function_imports.size()) {
```

Comment:
could use `!IsFunctionImport(...)` and remove the assert (since it's already in `IsFunctionImport`)